modules/porous_flow/src/actions/PorousFlowSinglePhaseBase.C

```diff
@@ -177,14 +198,34 @@ PorousFlowSinglePhaseBase::act()
   const bool compute_e_qp = _deps.dependsOn(_objects_to_add, "PorousFlowInternalEnergy_qp");
   const bool compute_h_qp = _deps.dependsOn(_objects_to_add, "PorousFlowEnthalpy_qp");
   if (compute_rho_mu_qp || compute_e_qp || compute_h_qp)
-    addSingleComponentFluidMaterial(false, 0, compute_rho_mu_qp, compute_e_qp, compute_h_qp, _fp);
+  {
+    if (_use_brine)
+    {
+      std::string nacl_name = _mass_fraction_vars[_nacl_index];
+      addBrineMaterial(nacl_name, false, 0, compute_rho_mu_qp, compute_e_qp, compute_h_qp);
+    }
+    else
+    {
+      addSingleComponentFluidMaterial(false, 0, compute_rho_mu_qp, compute_e_qp, compute_h_qp, _fp);
+    }
+  }
   const bool compute_rho_mu_nodal = _deps.dependsOn(_objects_to_add, "PorousFlowDensity_nodal") ||
                                     _deps.dependsOn(_objects_to_add, "PorousFlowViscosity_nodal");
   const bool compute_e_nodal = _deps.dependsOn(_objects_to_add, "PorousFlowInternalEnergy_nodal");
   const bool compute_h_nodal = _deps.dependsOn(_objects_to_add, "PorousFlowEnthalpy_nodal");
   if (compute_rho_mu_nodal || compute_e_nodal || compute_h_nodal)
-    addSingleComponentFluidMaterial(
-        true, 0, compute_rho_mu_nodal, compute_e_nodal, compute_h_nodal, _fp);
+  {
+    if (_use_brine)
+    {
+      std::string nacl_name = _mass_fraction_vars[_nacl_index];
+      addBrineMaterial(nacl_name, true, 0, compute_rho_mu_nodal, compute_e_nodal, compute_h_nodal);
```

Comment:
Please add a test similar to the one you've already added that checks enthalpy and internal energy too.

Reply:
Done - added to the existing test